.github/workflows/e2e-matrix.yml

```diff
@@ -53,6 +53,17 @@ jobs:
       ARTIFACTS: ${{ github.workspace }}/artifacts
 
     steps:
+    - name: Free disk space
```

Comment:
If I remember correctly, there is an action that does this, but seems good as well.
```yaml
      - name: Maximize build space
        uses: AdityaGarg8/remove-unwanted-software@90e01b21170618765a73370fcc3abbd1684a7793 # v5
        with:
          remove-android: 'true'
          remove-dotnet: 'true'
          remove-haskell: 'true'
          remove-codeql: 'true'
          remove-docker-images: 'false'
          remove-large-packages: 'true'
          remove-cached-tools: 'true'
          remove-swapfile: 'true'
```